### PR TITLE
115 matches api

### PR DESCRIPTION
**Pull request by TwoDigitsOneNumber**
implemented api according to google sheets https://docs.google.com/spreadsheets/d/1YOr29j0gQqa2Qz562hBfuy-4WK2rcYtldDeSnlX2MjU/edit#gid=0

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/controller/UserController.java [hunk 1]
```diff
@@ -60,6 +61,10 @@ public ResponseEntity<UserGetDTO> createUser(@RequestBody UserPostDTO userPostDT
     MultiValueMap<String, String> headers = new HttpHeaders();
     headers.set("token", createdUser.getToken());
 
+    System.out.println("created: " + createdUser.getName());
+    System.out.println(createdUser.getToken());
+    System.out.println(createdUser.getId());
+
```
Comment (daveking7) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/controller/UserController.java:67]: Didn't SonarLint complain already about that one?

**Review by daveking7 — APPROVED** [review]
Seems ok, but I would delete unnecessary printing statements before merging.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/controller/UserController.java (hunk 1 above)
Comment (TwoDigitsOneNumber) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/controller/UserController.java]: yes, you are right, i overlooked them

File: src/main/java/ch/uzh/ifi/hase/soprafs22/controller/UserController.java [hunk 2]
```diff
@@ -162,4 +167,22 @@ public void deleteUser(
...
       userService.deleteUser(userId); //Throws 404 if user with userId doesn't exist
   }
+
+  @GetMapping("/users/{id}/matches")
+  @ResponseStatus(HttpStatus.OK)
+  public List<UserGetDTO> getMatches(
+          @RequestHeader(value = "authorization", required = false) String token,
+          @PathVariable(value = "id") long id){
+
+      userService.checkSpecificAccess(token, id); // throws 401 if Access isn't allowed
```
[the remark this answers is not in the document]
Comment (TwoDigitsOneNumber) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/controller/UserController.java]: yes, i added the comment. but either way the correct error status should get returned. i just wanted to make sure both cases are covered

**Pull request merged**